test/e2e/route_advertisements.go

```diff
@@ -178,37 +180,217 @@ var _ = ginkgo.Describe("BGP: Pod to external server when default podNetwork is
+			if nodeIPv4 != "" {
+				clientPodNodeIPs = append(clientPodNodeIPs, nodeIPv4)
+			}
+			if nodeIPv6 != "" {
+				clientPodNodeIPs = append(clientPodNodeIPs, nodeIPv6)
+			}
+			framework.Logf("clientPodNodeIPs: %v", clientPodNodeIPs)
+
+			ginkgo.By("Creating a host networked pod on a second node")
+			// Add labels so the service can select this pod
+			podLabels := map[string]string{
+				"app": "internal-server-pod",
+			}
+
+			// Create host networked pod on another cluster node using createPod function
+			hostNetworkedPod, err = createPod(f, "internal-server-pod", nodes.Items[2].Name, f.Namespace.Name,
+				[]string{"/agnhost", "netexec", "--http-port=8080"}, podLabels, func(p *corev1.Pod) {
+					// Set host networking
+					p.Spec.HostNetwork = true
+
+					// Add required security context to comply with PodSecurity "restricted" policy
+					for i := range p.Spec.Containers {
+						if p.Spec.Containers[i].SecurityContext == nil {
+							p.Spec.Containers[i].SecurityContext = &corev1.SecurityContext{}
+						}
+
+						// Set required security context fields
+						p.Spec.Containers[i].SecurityContext.AllowPrivilegeEscalation = ptr.To(false)
+						p.Spec.Containers[i].SecurityContext.RunAsNonRoot = ptr.To(true)
+						p.Spec.Containers[i].SecurityContext.RunAsUser = ptr.To(int64(1000))
+						p.Spec.Containers[i].SecurityContext.Capabilities = &corev1.Capabilities{
+							Drop: []corev1.Capability{"ALL"},
+						}
+						p.Spec.Containers[i].SecurityContext.SeccompProfile = &corev1.SeccompProfile{
+							Type: corev1.SeccompProfileTypeRuntimeDefault,
+						}
+					}
+				})
+			framework.ExpectNoError(err)
+
+			hostNetworkedPodNode, err := f.ClientSet.CoreV1().Nodes().Get(context.TODO(), hostNetworkedPod.Spec.NodeName, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			if isIPv6Supported(f.ClientSet) {
+				nodeIPv6 = e2enode.GetAddressesByTypeAndFamily(hostNetworkedPodNode, corev1.NodeInternalIP, corev1.IPv6Protocol)[0]
+			}
+			if isIPv4Supported(f.ClientSet) {
+				nodeIPv4 = e2enode.GetAddressesByTypeAndFamily(hostNetworkedPodNode, corev1.NodeInternalIP, corev1.IPv4Protocol)[0]
+			}
+			hostNetworkedPodNodeIPs = append(hostNetworkedPodNodeIPs, nodeIPv4)
+			hostNetworkedPodNodeIPs = append(hostNetworkedPodNodeIPs, nodeIPv6)
+			framework.Logf("hostNetworkedPodNodeIPs: %v", hostNetworkedPodNodeIPs)
+
+			ginkgo.By("With default network being advertised, queries to the external server are not SNATed (uses podIP)")
+			success := testClientPodToServerConnectivity(serverContainerIPs, "8080", clientPodIPs, "external server pod")
+			if !success {
+				ginkgo.Fail("With default network being advertised initially, pod to external server test failed")
+			}
+
+			ginkgo.By("With default network being advertised, queries to the second node are SNATed (uses nodeIP)")
+			success = testClientPodToServerConnectivity(hostNetworkedPodNodeIPs, strconv.Itoa(8080), clientPodNodeIPs, "client pod to second node")
+			if !success {
```

Comment:
Please consider to use ginkgo.Expect() instead.

Reply:
updated